src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+        if xindexL[j] >= 0:
+            if xindexLB[j - 1] >= 0.0:
+                s = int(math.floor(xindexLB[j - 1]))
+                e = int(math.floor(xindexLB[j]))
+
+                col[:] = (
+                    (s + 1 - xindexLB[j - 1]) * imgin[:, s]
+                    + (xindexLB[j] - e) * imgin[:, e]
+                )
+
+                if (e - s) > 1:   # have a middle pixel
+                    col[:] = col[:] + imgin[:, s + 1]
+
+                # Perform the desired noise reduction method
+                col = noise_reduce(col, xtable, j, xindexLB, noise_reduction)
+
+                # TODO: Check on this. Make sure it wasn't an
+                # error in the original code
+                if 0 != noise_reduction:
+                    low_mask = (col < 0)
+                    col[low_mask] = 0  # underflow?
+
+                high_mask = (col > maxlevel)
+                col[high_mask] = maxlevel  # saturated?  for max image==1.0
+
+                imgout[:, j] = col
+            else:
+                imgout[:, j] = imgin[:, xindexL[j]] * xtable['bgfactorL']
+        else:
+            imgout[:, j] = xtable['mean_modevalue'] * xtable['bgfactorL']
+
+    # Right side
+    for j in range(0, int(xtable['aR'])):
+        col[:] = 0.0
+        if xindexR[j] >= 0:
+            if xindexRB[j - 1] >= 0.0:
+                s = int(math.floor(xindexRB[j - 1]))
+                e = int(math.floor(xindexRB[j]))
+
+                col[:] = (
+                    (s + 1 - xindexRB[j - 1]) * imgin[:, 511 - s]
+                    + (xindexRB[j] - e) * imgin[:, 511 - e]
+                )
+
+                if (e-s) > 1:  # have a middle pixel
+                    col[:] = col[:] + imgin[:, 511 - (s + 1)]
+
+                # Perform the desired noise reduction method
+                col = noise_reduce(col, xtable, j, xindexRB, noise_reduction)
+
+                # TODO: Check on this. Make sure it wasn't an
+                # error in the original code
+                if 0 != noise_reduction:
+                    low_mask = (col < 0)
+                    col[low_mask] = 0  # underflow?
+
+                high_mask = (col > maxlevel)
+                col[high_mask] = maxlevel  # saturated?  for max image==1.0
+
+                imgout[:, 511 - j] = col
+            else:
+                imgout[:, 511 - j] = (
+                    imgin[:, 511 - xindexR[j]]
+                    * xtable['bgfactorR']
+                )
+        else:
+            imgout[:, 511 - j] = xtable['mean_modevalue'] * xtable['bgfactorR']
+
+    if FOVwidth == 512:
+        img = imgout.astype(np.uint16)
+    else:
+        img = imgout[:, xtable['L']:512 - xtable['R']].astype(np.uint16)
+
+    return img
+
+
+def get_xindex(a: float, b: float):
+    """
+    Generate the xindex arrays that will be used in the dewarping process.
+
+    Parameters
+    ----------
+    a: float
+        Information about the rig used for the experiment.
```

Comment:
do you have anything more informative to say about these arguments? these aren't helpful to understand what this is.